### PR TITLE
PYTHON-2595 Raise PyMongo exceptions from PyMongoArrow where appropriate

### PR DESCRIPTION
**Pull request by prashantmital**

### CODE REVIEW

File: bindings/python/pymongoarrow/errors.py [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+# limitations under the License.
+
+"""Exceptions raised by PyMongoArrow."""
+
+from pymongo.errors import *
+
+
+class PyMongoArrowError(Exception):
+    """Base class for all PyMongoArrow exceptions."""
+    pass
```
Comment (ShaneHarvey) [inline bindings/python/pymongoarrow/errors.py:22]: Newline

File: bindings/python/pymongoarrow/errors.py [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Exceptions raised by PyMongoArrow."""
+
+from pymongo.errors import *
```
Comment (ShaneHarvey) [inline bindings/python/pymongoarrow/errors.py:17]: Can you change this to explicitly import what we need? It makes it easier to track what's part of the api.

File: bindings/python/pymongoarrow/lib.pyx [hunk 3]
```diff
@@ -25,6 +25,7 @@ import enum
 import numpy as np
 from pyarrow import timestamp
 from pymongoarrow.types import _BsonArrowTypes
+from pymongoarrow.errors import *
```
Comment (ShaneHarvey) [inline bindings/python/pymongoarrow/lib.pyx:28]: Can you change this to explicitly import what we need? It makes it easier to track what's part of the api.

File: bindings/python/pymongoarrow/errors.py (hunk 1 above)
Comment (prashantmital) [reply bindings/python/pymongoarrow/errors.py]: Done

File: bindings/python/pymongoarrow/errors.py (hunk 2 above)
Comment (prashantmital) [reply bindings/python/pymongoarrow/errors.py]: Done.

File: bindings/python/pymongoarrow/lib.pyx (hunk 3 above)
Comment (prashantmital) [reply bindings/python/pymongoarrow/lib.pyx]: Done.

File: bindings/python/pymongoarrow/errors.py [hunk 4]
```diff
@@ -0,0 +1,22 @@
...
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Exceptions raised by PyMongoArrow."""
+
+from pymongo.errors import InvalidBSON
```
Comment (ShaneHarvey) [inline bindings/python/pymongoarrow/errors.py:17]: This only works by accident, you should import InvalidBSON from bson instead:
`from bson.errors import InvalidBSON`

Comment (prashantmital) [reply bindings/python/pymongoarrow/errors.py]: Done.

**Review by ShaneHarvey — APPROVED** [review]

**Pull request merged**